pytorch_lightning/callbacks/lr_logger.py

```diff
@@ -63,22 +77,33 @@ def on_train_start(self, trainer, pl_module):
         # Initialize for storing values
         self.lrs = {name: [] for name in names}
 
-    def on_train_batch_start(self, trainer, pl_module, batch, batch_idx, dataloader_idx):
-        latest_stat = self._extract_lr(trainer, 'step')
-        if trainer.logger and latest_stat:
-            trainer.logger.log_metrics(latest_stat, step=trainer.global_step)
+    def on_batch_start(self, trainer, pl_module):
+        if self.logging_interval is None:
+            latest_stat = self._extract_lr(trainer, 'step')
+            if trainer.logger and latest_stat:
+                trainer.logger.log_metrics(latest_stat, step=trainer.global_step)
+        elif self.logging_interval == 'step':
+            latest_stat = self._extract_lr(trainer, 'any')
+            if trainer.logger and latest_stat:
+                trainer.logger.log_metrics(latest_stat, step=trainer.global_step)
```

Comment:
double checking, is it really None -> step and step -> any?

Reply:
if `logging_interval` is set to` None`, then only `lr` that updates every step will be extracted,` lr` that updates every epoch will be skipped, finally we log extracted lr every step,
if set to `'step'`, lr that updates every step and every epoch will be both extracted, finally we log` lr` every step.
similarly for `epoch`.

Reply:
see https://github.com/PyTorchLightning/pytorch-lightning/pull/2847#issuecomment-670557150

Reply:
Ok, thanks for explaining. I think I understand. I guess the confusing part here was that I was just looking at on_batch_start but the other half of the logging is happening in on_epoch_start. 
Nonetheless I have a suggestion regarding the code. The way I would have implemented it is defining a mapping
```python
mapping = {None: "step", "step": any} 
```
and then get rid of the duplicated code in the if blocks and just do:
```python
extract_key = mapping[self.logging_interval]
latest_stat = self._extract_lr(trainer, extract_key)
            if trainer.logger and latest_stat:
                trainer.logger.log_metrics(latest_stat, step=trainer.global_step)

```
and analogous in epoch_end method. What do you think of this?

Reply:
It looks better, could you commit the changes.

Reply:
sorry no, I don't have access to your branch to push changes.

Reply:
I don't think this will work because `on_batch_start` will be called even when logging_interval == 'epoch'. let me try something else.

Reply:
does your reply refer to my suggestion or the initial question I had in this thread?

Reply:
your suggestion

Reply:
although I realized later that your suggestion will work with an update. made those changes :)

Reply:
ok, maybe I made a typo somewhere, but the idea was to make it a refactor, so equivalent code. 
anyway I think it is strange that you had to make a change and the tests did not fail? Doesn't that mean the test is incomplete?

Reply:
ok my bad, not enough sleep 😅